### ISSUE TITLE
Ranger crashes under python 3.10.1 

### ISSUE THREAD

**Issue by matthiasantierens** [ranger/ranger#2524]
#### Runtime Environment

- Operating system and version: Arch 5.15.8-arch1-1
- Terminal emulator and version: alacritty 0.9.0
- Python version: 3.10.1 (main, Dec 11 2021, 17:22:55) [GCC 11.1.0]
- Ranger version/commit: ranger 1.9.3
- Locale: en_US.UTF-8

#### Current Behavior
Ranger crashes by going into different dirs and files. (was not an issue before python update on Arch)

#### Context
Arch had a python update and some packages are broken.

#### Traceback

```
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/ranger/core/main.py", line 201, in main
    fm.loop()
  File "/usr/lib/python3.10/site-packages/ranger/core/fm.py", line 377, in loop
    ui.redraw()
  File "/usr/lib/python3.10/site-packages/ranger/gui/ui.py", line 333, in redraw
    self.draw()
  File "/usr/lib/python3.10/site-packages/ranger/gui/ui.py", line 360, in draw
    DisplayableContainer.draw(self)
  File "/usr/lib/python3.10/site-packages/ranger/gui/displayable.py", line 260, in draw
    displayable.draw()
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/view_miller.py", line 100, in draw
    DisplayableContainer.draw(self)
  File "/usr/lib/python3.10/site-packages/ranger/gui/displayable.py", line 260, in draw
    displayable.draw()
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/browsercolumn.py", line 187, in draw
    self._draw_file()
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/browsercolumn.py", line 214, in _draw_file
    Pager.draw(self)
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/pager.py", line 109, in draw
    self._draw_line(i, line)
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/pager.py", line 139, in _draw_line
    self.set_fg_bg_attr(*chunk)
  File "/usr/lib/python3.10/site-packages/ranger/gui/curses_shortcuts.py", line 89, in set_fg_bg_attr
    self.win.attrset(curses.color_pair(get_color(fg, bg)) | attr)
  File "/usr/lib/python3.10/site-packages/ranger/gui/color.py", line 32, in get_color
    curses.init_pair(size, fg, bg)
ValueError: Color number is greater than COLORS-1 (7).
```

**Comment by chu4ng** [comment]
I'm using arch too. My python and alacritty versions are same, but everythink works fine.
Can you send your ncurses and curses(from python) version?
`python3.10 -c "import curses; print(curses.version)"  `
`pacman -Qs ncurses`

**Comment by matthiasantierens** [comment]
curses `version b'2.2'`

```
pacman -Qs ncurses
local/lib32-ncurses 6.3-1
    System V Release 4.0 curses emulation library (32-bit)
local/ncdu 1.16-2
    Disk usage analyzer with an ncurses interface
local/ncurses 6.3-1
    System V Release 4.0 curses emulation library
    ```

Just set my TERM var and it seems stable.
Thank you for the help!

**Comment by arobase-che** [comment]
Same here. Using `xterm -ti vt340 -e 'ranger'`. On an almost exact same environment as yours.

My personal color-schemes hides the problem for some time but using the default ranger configuration made it happens on almost every preview. So if you want to reproduce, you may be able to using `xterm` and by deleting `~/.config/ranger`.

`xterm -ti vt340 -e 'TERM=xterm-256color ranger'` is a effective fix.

I'm not able to reproduce the bug on alacritty nor on kitty.

**Comment by nippyin** [comment]
Same error on macOS + Kitty 

I see this error when going through folder and it contains dot files.


ranger version: ranger 1.9.3
Python version: 3.10.1 (main, Dec  6 2021, 22:25:40) [Clang 13.0.0 (clang-1300.0.29.3)]
Locale: en_CA.UTF-8
Current file: '/Users/myuser/code/Azure/.funcignore'

Traceback (most recent call last):
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/core/main.py", line 201, in main
    fm.loop()
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/core/fm.py", line 377, in loop
    ui.redraw()
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/ui.py", line 333, in redraw
    self.draw()
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/ui.py", line 360, in draw
    DisplayableContainer.draw(self)
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/displayable.py", line 260, in draw
    displayable.draw()
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/widgets/view_miller.py", line 100, in draw
    DisplayableContainer.draw(self)
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/displayable.py", line 260, in draw
    displayable.draw()
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/widgets/browsercolumn.py", line 187, in draw
    self._draw_file()
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/widgets/browsercolumn.py", line 214, in _draw_file
    Pager.draw(self)
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/widgets/pager.py", line 109, in draw
    self._draw_line(i, line)
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/widgets/pager.py", line 139, in _draw_line
    self.set_fg_bg_attr(*chunk)
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/curses_shortcuts.py", line 89, in set_fg_bg_attr
    self.win.attrset(curses.color_pair(get_color(fg, bg)) | attr)
  File "/usr/local/Cellar/ranger/1.9.3_2/libexec/ranger/gui/color.py", line 32, in get_color
    curses.init_pair(size, fg, bg)
ValueError: Color number is greater than COLORS-1 (7).

ranger crashed. Please report this traceback at:
https://github.com/ranger/ranger/issues

**Comment by Haptein** [comment]
I'm getting the exact same traceback (apart from the line numbers bc different ranger version), it doesn't happen on every file. Here's a device path where it happens, maybe that can help with reproducing it. Tested with xterm and screen as TERM and both gave the same problem.

Manjaro with cool-retro-term v1.1.1
ranger version: ranger-master
Python version: 3.10.1 (main, Dec 18 2021, 23:53:45) [GCC 11.1.0]
Locale: en_GB.UTF-8
Current file: '/sys/module/nvme/drivers/pci:nvme/0000:02:00.0/power/control'

Setting TERM to xterm-256color did fix the problem for me.

**Comment by seanwarman** [comment]
Setting TERM to xterm-256color worked to fix the problem for me. 

Here's some inbfo in case it helps with the problem...

I get the error when using ranger in tmux and iTerm on macos. If I run ranger using tmux's `:split ranger` command, it'll run ranger in zsh without running zshrc and I get this crash error but only when I try to preview files. It'll work traversing directories.

I fixed it by running the following in tmux instead: `:split zsh -c "export TERM=xterm-256color && ranger"`

**Comment by nadapez** [comment]
The problem is that the exception is supposed to be catched but  the `try` clause expects a `curses.error` exception which is different from the actual exception raised: `ValueError`.

I suppose is due to a change in `curses` package.

Solution for version 1.9.3: change the lines #32 and #44 in `color.py`  from `except curses.error:` to `except (curses.error, ValueError)`:

**Comment by kjeldonarch** [comment]
I don't know, what may be the reason, just want to share my traceback.
Ranger was installed yesterday, I was on TTY2 (bash), preview active (also for images, because i changed scope.sh to check for active X -- if it's not available, preview_images is set to false (only locally in the script).
Ranger crashed, when opening the direction.

System: Arch ; full encrypted 
terminal: bash
no personal color settings in Ranger

```
ranger version: ranger 1.9.3
Python version: 3.10.6 (main, Aug  3 2022, 17:39:45) [GCC 12.1.1 20220730]
Locale: de_DE.UTF-8
Current file: '/usr/share/ImageMagick-7/english.xml'

Traceback (most recent call last):
  
    fm.loop()
  File "/usr/lib/python3.10/site-packages/ranger/core/fm.py", line 377, in loop
    ui.redraw()
  File "/usr/lib/python3.10/site-packages/ranger/gui/ui.py", line 333, in redraw
    self.draw()
  File "/usr/lib/python3.10/site-packages/ranger/gui/ui.py", line 360, in draw
    DisplayableContainer.draw(self)
  File "/usr/lib/python3.10/site-packages/ranger/gui/displayable.py", line 260, in draw
    displayable.draw()
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/view_miller.py", line 100, in draw
    DisplayableContainer.draw(self)
  File "/usr/lib/python3.10/site-packages/ranger/gui/displayable.py", line 260, in draw
    displayable.draw()
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/browsercolumn.py", line 187, in draw
    self._draw_file()
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/browsercolumn.py", line 214, in _draw_file
    Pager.draw(self)
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/pager.py", line 109, in draw
    self._draw_line(i, line)
  File "/usr/lib/python3.10/site-packages/ranger/gui/widgets/pager.py", line 139, in _draw_line
    self.set_fg_bg_attr(*chunk)
  File "/usr/lib/python3.10/site-packages/ranger/gui/curses_shortcuts.py", line 89, in set_fg_bg_attr
    self.win.attrset(curses.color_pair(get_color(fg, bg)) | attr)
  File "/usr/lib/python3.10/site-packages/ranger/gui/color.py", line 32, in get_color
    curses.init_pair(size, fg, bg)
ValueError: Color number is greater than COLORS-1 (7).

ranger crashed. Please report this traceback at:
https://github.com/ranger/ranger/issues
```

**Comment by jmadison222** [comment]
TLDR:
```
export TERM=xterm-256color
```
It says that in the thread above, but it was hard to find.  Just making it easy to find.
